### DISCUSSION TITLE
Nested ternaries

### DISCUSSION THREAD

**Discussion by Arthurvdv** [StefanMaron/BusinessCentral.LinterCop#713] [New Rule]
An interesting [discussion on X/Twitter](https://x.com/ajkauffmann/status/1813608707965743270) on the new ternary operator in Business Central v25.

Unsure of this is feasible, but a diagnostic on nested ternaries in the context of (un)readable code.

**Comment by rvanbekkum** [comment]
As in, avoid nested ternaries completely? Or, a warning for nested ternaries starting from a certain 'nesting level' (with a setting in LinterCop.json)?
I think you should avoid a nested ternary operator completely, but I can also imagine that others have a different opinion on that, so it should be good to take that into account.

**Reply by Arthurvdv** [reply]
Currently I would opt for “.. avoid nested ternaries complete….”, but it’s a bit early to already decide right now. I wanted to create the discussion here, while I still had de link to the twitter thread.

**Reply by Arthurvdv** [reply]
[The Dark Side of Ternary Operator](https://x.com/VDvernytskyi/status/1814228232469672232)
_(gathering more content here, now the discussion on this is on topic)_

**Comment by Arthurvdv** [comment]
With the new [LC0090 - Show Cognitive Complexity diagnostics for methods above threshold](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0090) rule, the Cognitive Complexity metric will cover this.

```AL
procedure GetDayName(Day: Integer): Text   // Cognitive Complexity: 28 (threshold >=15)
var
    DayName: Text;
begin
    DayName := (Day = 1) ? 'Monday' :    // +1 (nested = 0)
            (Day = 2) ? 'Tuesday' :      // +2 (nested = 1)
            (Day = 3) ? 'Wednesday' :    // +3 (nested = 2)
            (Day = 4) ? 'Thursday' :     // +4 (nested = 3)
            (Day = 5) ? 'Friday' :       // +5 (nested = 4)
            (Day = 6) ? 'Saturday' :     // +6 (nested = 5)
            (Day = 7) ? 'Sunday' :       // +7 (nested = 6)
            'Invalid Day';
    exit(DayName);
end;

procedure GetDayName(Day: Integer): Text   // Cognitive Complexity: 1 (threshold >=15)
begin
    case Day of                          // +1 (nested = 0)
        1:
            exit('Monday');
        2:
            exit('Tuesday');
        3:
            exit('Wednesday');
        4:
            exit('Thursday');
        5:
            exit('Friday');
        6:
            exit('Saturday');
        7:
            exit('Sunday');
        else
            exit('Invalid Day');
    end;
end;
```
